core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -146,6 +185,10 @@ class TaskMetrics extends Serializable {
     }
     _shuffleReadMetrics = Some(merged)
   }
+
+  private[spark] def updateInputMetrics() = synchronized {
```

Comment:
So this follows the method above: updateShuffleReadMetrics that doesn't have a return type. Should I change both then?